### PR TITLE
Prove spthcycl

### PR DESCRIPTION
**Pull request by BTernaryTau**
> A walk is a trivial path if and only if it is both a simple path and a cycle.

### CODE REVIEW

**Review by tirix — APPROVED** [review]
Looks good!
I would be good to wait for @avekens's approval too, though.

**Review by avekens — APPROVED** [review]

**Review by icecream17 — APPROVED** [review]

**Pull request merged**
